### ISSUE TITLE
Extract out duplicated code in FeedbackMcqQuestionDetails.java

### ISSUE THREAD

**Issue by wenmogu** [TEAMMATES/teammates#8244]
**Description of feature/enhancement**
There are two occurrences of duplicated codes in FeedbackMcqQuestionDetails.java

![image](https://user-images.githubusercontent.com/28526038/34321634-caefb268-e84e-11e7-98d3-96e0ec03477a.png)

![image](https://user-images.githubusercontent.com/28526038/34321636-e311d5d8-e84e-11e7-87c7-512179197feb.png)

**Justification**
Duplicated code fragments can be extracted out into a single method and reused in different parts. This reduces the length of the code and improves readability.

**Comment by damithc** [comment]
@wenmogu are they actual duplicates that seem  to be extract-able? IDEs can warning can be false positives.

**Comment by wkurniawan07** [comment]
@wenmogu I'm a firm believer of the DRY principle myself, but be careful when concluding that "Duplicated code fragments can be extracted out into a single method". When you're extracting something out, that abstraction has to make sense on its own and not just to simply "reduce duplication". There are times where it's better to have duplicated codes than wrong abstractions.

**Comment by wenmogu** [comment]
@damithc @wkurniawan07 Thank you for your inputs. Below r my reasoning in posting this issue.

The first pic shows the duplicated code that exists in method `getQuestionResultStatisticsCsv` and method `getQuestionResultStatisticsHtml` in class `FeedbackMcqQuestionDetails`. This code fragment is for collation of answering frequencies in all the responses to the MCQ questions, on whether the response is the answer or "other". Based on my understanding, since this is the only use of the fragment and this fragment is used multiple times, it is better to extract it out as a private method inside the class and reused. I have tried to do the abstraction and it doesnt break any test. 

The fragment shown in the second pic has its sister version in class `FeedbackMsqQuestionDetails`, method `generateOptionList` (same method name as in class `FeedbackMcqQuestionDetails`). I have to admit that they are not exactly the same as they involve private attributes from each class. I tried the abstraction by extracting the method `generateOptionList` into their parent class `FeedbackQuestionDetails`. Although the tests dont break, I think it is not a good idea as no polymorphism is observed for the methods in that class except for the abstract methods.
